---
title: "miR Enrichment"
author: "Burnet Bioinformatics"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
    fig_width: 7
    fig_height: 7
theme: cosmo
---

Source: https://github.com/markziemann/miR-enrichment

## Introduction

The predicted targets of microRNAs are used in functional enrichment analysis to justify the
potential function of microRNAs, but there are some logical problems with this.
In a biological tissue, not all of these targets will be expressed.
Also the enrichment analysis requires a background list which is all the genes that can be
measured.
Not all genes are expressed in a tissue, at least half are silenced, so each enrichment
analysis requires a custom background gene list.
Unfortunately a custom background list is rarely used.
We argue that this causes a dramatic distortion to the results.

## Sample sheet

GSE232687

Control mRNA: Mock_IN_1 Mock_IN_2 Mock_IN_3

Case mRNA: RSV_IN_1 RSV_IN_2 RSV_IN_3

## Methods


```{r,packages}

suppressPackageStartupMessages({
    library("DESeq2")
    library("gplots")
    library("mitch")
    library("eulerr")
    library("getDEE2")
    library("kableExtra")
})

```

## Import read counts

Importing RNA-seq data

```{r,importdata}

x <- read.table("GSE231788_SR92_genes_n_regRegions_raw_counts_for_degust_GEO.csv.gz",sep=",",header=TRUE)
x$gene <- paste(x$gene_id,x$gene_symbol)
x$biotype = x$chr = x$gene_symbol = x$gene_id = NULL
xa <- aggregate(. ~ gene, x , sum)
dim(x) ; dim(xa)
rownames(xa) <- xa$gene
xa$gene = NULL
table(grepl("ENSR",rownames(xa)))
xarat <- xa[grep("ENSR",rownames(xa)),]
colSums(xarat)
xah <- xa[grep("ENSG",rownames(xa)),]
colSums(xah)
head(xah)

xahf <- xah[which(rowMeans(xah)>=10),]
dim(xah) ; dim(xahf)

```

## Differential expression

```{r,de}

ss <- data.frame("run"=colnames(xahf),"trt"=c(0,0,0,1,1,1))
rownames(ss) <- colnames(xahf)

mds <- cmdscale(dist(t(xahf)))
plot(mds,cex=2,col="gray",pch=19)
text(mds, labels=rownames(mds) ,col="black")

colSums(xahf)

dds <- DESeqDataSetFromMatrix(countData = round(xahf) , colData = ss , design = ~ trt )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <- cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
head(dge)
ups <- rownames(subset(dge,padj<0.05 & log2FoldChange>0))
dns <- rownames(subset(dge,padj<0.05 & log2FoldChange<0))
lapply(list("UPs"=ups,"DNs"=dns),length)
nrow(dge)

saveRDS(dge,file="GSE232687.Rmd")

```

## Session information

For reproducibility.

```{r,session}

sessionInfo()

```

